l10n_it_vat_registries/models/vat_registry.py

```diff
@@ -0,0 +1,210 @@
+            "registry_type": data["form"]["registry_type"],
+            "invoice_total": self._get_move_total,
+            "tax_registry_name": data["form"]["tax_registry_name"],
+            "env": self.env,
+            "formatLang": formatLang,
+            "compute_totals_tax": self._compute_totals_tax,
+            "l10n_it_count_fiscal_page_base": data["form"]["fiscal_page_base"],
+            "only_totals": data["form"]["only_totals"],
+            "date_format": date_format,
+            "year_footer": data["form"]["year_footer"],
+        }
+        return docargs
+
+    def _get_move(self, move_ids):
+        move_list = self.env["account.move"].browse(move_ids)
+        return move_list
+
+    def _format_date(self, my_date, date_format):
+        # supporting both cases, as data['form']['from_date'] is string
+        if isinstance(my_date, str):
+            formatted_date = time.strftime(
+                date_format, time.strptime(my_date, "%Y-%m-%d")
+            )
+        else:
+            formatted_date = my_date.strftime(date_format)
+        return formatted_date or ""
+
+    def _get_invoice_from_move(self, move):
+        return self.env["account.move"].search([("id", "=", move.id)])
+
+    def _get_move_line(self, move, data):
+        return [move_line for move_line in move.line_ids]
+
+    def _tax_amounts_by_tax_id(self, move, move_lines, registry_type):
+        res = {}
+
+        for move_line in move_lines:
+            set_cee_absolute_value = False
+            if not (move_line.tax_line_id or move_line.tax_ids):
+                continue
+
+            if move_line.tax_ids and len(move_line.tax_ids) != 1:
+                raise UserError(
+                    _("Move line %s has too many base taxes") % move_line.name
+                )
+
+            if move_line.tax_ids:
+                tax = move_line.tax_ids[0]
+                is_base = True
+            else:
+                tax = move_line.tax_line_id
+                is_base = False
+
+            if (registry_type == "customer" and tax.cee_type == "sale") or (
+                registry_type == "supplier" and tax.cee_type == "purchase"
+            ):
+                set_cee_absolute_value = True
+
+            elif tax.cee_type:
+                continue
+
+            if tax.parent_tax_ids and len(tax.parent_tax_ids) == 1:
+                # we group by main tax
+                tax = tax.parent_tax_ids[0]
+
+            if tax.exclude_from_registries:
+                continue
+
+            if not res.get(tax.id):
+                res[tax.id] = {
+                    "name": tax.name,
+                    "base": 0,
+                    "tax": 0,
+                }
+            tax_amount = move_line.debit - move_line.credit
+
+            if set_cee_absolute_value:
+                tax_amount = abs(tax_amount)
+
+            if "receivable" in move.move_type or "payable_refund" == move.move_type:
```

Comment:
Non ho trovato `payable_refund` tra i possibili valori di `move_type`, puoi verificare?

Reply:
il campo **move_type** in 12.0 erà inserito dal modulo **account_tax_balance** ... ora è diventato **financial_type**. modifico, grazie

Reply:
@SimoRubi anche le altre richieste dovrei averle risolte, puoi controllare? grazie